src/GitVersionCore.Tests/IntegrationTests/HotfixBranchScenarios.cs

```diff
@@ -63,7 +63,7 @@ public void CanTakeVersionFromHotfixesBranch()
 
             // create hotfix branch
             Commands.Checkout(fixture.Repository, fixture.Repository.CreateBranch("hotfixes/1.1.1"));
-            fixture.AssertFullSemver("1.1.0"); // We are still on a tagged commit
+            fixture.AssertFullSemver("1.1.1-beta.1+0"); // Version in branch name takes precedence over Tag if it's greater
```

Comment:
Hm. I'm torn and don't really know what to think here. My knee-jerk reaction is that tags should always take precedence over everything else, but then again the entire point with versioned `hotfix` and `release` branches is to affect the version number.

Reply:
I think it makes sense that even if we are on tagged commit on release branch, greater version wins between tag and branch name.
As at first Release branch for X version is just created and optionally some Hotfixes are provided. 
But the whole release process on such branch may end with no additional commits. 

In the end I just documented what I experienced when I had to update this test case scenario.

Reply:
Actually I agree with 1.1.1-beta.1+0 as a result for this scenario. If you think about it: You are going to create hotfix branch from the current tagged version from main (RTM version). Thus your plan is to fix something maybe your CI pipeline has changes or what ever. The point is: This version says here that something different is included comparing to the previous version 1.1.0 but no code changes are involved (beta.1+0). ;)

Reply:
Maybe you have changed from x64 to x32 or used the self-contained flag in dotnet because on your production environment dotnet is not installed.